mappers/bluetooth_mapper/README.md

```diff
@@ -0,0 +1,311 @@
+# Bluetooth Mapper
+
+
+## Introduction
+
+Mapper is an application that is used to connect  and control devices. This is an implementation of mapper for 
+bluetooth protocol. The aim is to create an application through which users can easily operate  devices using bluetooth protocol
+for communication to the KubeEdge platform. The user is required to provide the mapper with the information required to control their device
+through the configuration file these can be changed at runtime by providing the input through the MQTT broker.
+
+
+## Modules
+
+There are five modules in this implementation of bluetooth mapper, namely :
+ 1. Action Manager
+ 2. Scheduler
+ 3. Watcher
+ 4. Data Converter
+ 5. Controller
+ 
+ ### Action Manager
+ 
+ A device can be controlled by setting a specific value in physical register(s) of a device and readings can be acquired by getting the value from specific register(s).
+ We can define an Action as a group of read/write operations on a device. A device may support multiple such actions. Each of these actions should be supplied through config-file to action manager
+ or through MQTT messages received at runtime. The user needs to provide the mapper with the information about the registers by
+ providing the Characteristic-UUID for each action to be performed. The following is the structure used by the action manager  module :
+ 
+ ```go
+// Operation is structure to define device operation
+type Operation struct {
+	// Action can be one of read/write corresponding to get/set respectively
+	Action string `yaml:"action" json:"action"`
+	// Characteristic is  refers to the characteristic on which the operation needs to be performed
+	CharacteristicUUID string `yaml:"characteristic-uuid" json:"characteristic-uuid"`
+	// Value is the value to be written in case of write action and value read from the device in case of read action
+	Value []byte `yaml:"value" json:"value"`
+}
+
+// Action is structure to define a device action
+type Action struct {
+	Enable bool `yaml:"enable" json:"enable"`
+	// Name is the name of the Action
+	Name string `yaml:"name" json:"name"`
+	// Operation specifies the operation to be performed for this action
+	Operation Operation `yaml:"operation" json:"operation"`
+}
+
+//ActionManager is a structure that contains a list of actions
+type ActionManager struct {
+	Actions []Action `yaml:"actions"`
+}
+```
+
+ ### Scheduler
+ 
+  Scheduler is used to perform action(s) after defined intervals. A limit can be set for the number for the number of times an action 
```

Comment:
nit: "for the number" occurs twice. And please remove unneeded whitespaces and move lines up.

Reply:
done, thanks